### PR TITLE
Integrate live data with historic

### LINKED ISSUE #53: Coinbase Data aggregation
Aggregate data from historic and live Coinbase data sources.

### PR DESCRIPTION
**Pull request by CBarrett2**
Resolves #53

### CODE REVIEW

File: src/assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js [hunk 1]
```diff
@@ -7,8 +7,8 @@
         var period = 60 * 60 * 24;
         var liveFeed = sc.data.feed.coinbase.webSocket();
 
-        function ohlc(cb) {
-            var basket = null;
+        function ohlc(cb, initialBasket) {
```
Comment (janakerman) [inline src/assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js]: Could you rename this to ohlcWebSocketAdaptor? I know you didn't change it in this commit, but it'll be a quick change and it doesn't feel worth raising as an issue to fix later.

Comment (CBarrett2) [inline src/assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js]: Yeah, not a problem!

File: src/assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js [hunk 2]
```diff
@@ -7,8 +7,8 @@
...
+            var basket = initialBasket || null;
```
Comment (janakerman) [inline src/assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js]: `null == undefined` is `true`, I don't think there's any need to differentiate between the two in this scenario. I'd just put
    
    var basket = intialBasket;

File: src/assets/js/main.js [hunk 3]
```diff
@@ -119,6 +119,32 @@
...
+        if (data[data.length - 1].date.getTime() !== basket.date.getTime()) {
+            data.push(basket);
+        } else {
+            data[data.length - 1] = basket;
+        }
+        render();
+    }
+
+    function liveCallback(event, latestBasket) {
+        if (!event && latestBasket) {
```
Comment (janakerman) [inline src/assets/js/main.js]: Is the lack of an event parameter being used to specify that there was an update event? Might be nicer to add an event type, unless these events are defined by the socket protocol or something, then I'd rename this parameter `socketEvent` or something the like. Just to make it a little more clear that it's an event relating to sockets, and not just to your adaptor.

File: src/assets/js/main.js [hunk 4]
```diff
@@ -119,6 +119,32 @@
...
+            newBasketReceived(latestBasket);
+        } else if (event.type === 'open') {
```
Comment (janakerman) [inline src/assets/js/main.js]: I'd just delete the empty branches of this if statement.

File: src/assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js [hunk 5]
```diff
@@ -7,8 +7,8 @@
         var period = 60 * 60 * 24;
         var liveFeed = sc.data.feed.coinbase.webSocket();
 
-        function ohlc(cb) {
-            var basket = null;
+        function ohlcWebSocketAdaptor(cb, initialBasket) {
```
Comment (CBarrett2) [inline src/assets/js/data/feed/coinbase/ohlcWebSocketAdaptor.js]: Should I pass in period here, rather than hold it as a state? re: our discussion about unit tests?

File: src/assets/js/main.js (hunk 4 above)
Comment (CBarrett2) [inline src/assets/js/main.js]: I'd have to then check that event !== 'open && !(event === 'close' && event.code === 1000) in the last check then? Not sure if there's a better way

File: src/assets/js/main.js [hunk 6]
```diff
@@ -119,6 +119,34 @@
...
+        }
+        render();
+    }
+
+    function liveCallback(socketEvent, latestBasket) {
+        if (latestBasket) {
+            newBasketReceived(latestBasket);
+        }
+
+        if (socketEvent) {
```
Comment (janakerman) [inline src/assets/js/main.js]: I wouldn't nest the IF..  (socketEvent && !sucessfulOpen && !successfulClose) just because I always find deep logic nesting hard to follow (and these things always tend to grow!) but now I'm being super picky!

Comment (CBarrett2) [inline src/assets/js/main.js]: The problem is that sometimes socketEvent is null so it will throw an error if I try to check socketEvent.type and socketEvent is null. I could use, say `var successfulClose = socketEvent && socketEvent.type === 'close' && socketEvent.code === 1000; perhaps
